### PR TITLE
refactor(deposit): forceApprove

### PR DESCRIPTION
**Pull request by nknavkal**
- Replace direct .approve() with .forceApprove() via SafeERC20 in all three derived deposit address contracts; move SafeERC20 import and using declaration from BaseDepositAddress into each derived contract where it is actually used
- Remove unused SafeERC20 import and using declaration from BaseDepositAddress
- Remove unused error InvalidDestinationPortal() from BaseDepositFactory
- Remove redundant explicit return statements on named return variables in BaseDepositAddress.createIntent(), BaseDepositFactory.deploy(), and all three _executeIntent() implementations
- Remove redundant inline comment in DepositFactory_USDCTransfer_Solana._getSalt()
- Add missing blank line after contract opening brace in DepositFactory_CCTPMint_Arc

## Pull Request Title Format

Please ensure your PR title follows the [Conventional Commit](https://www.conventionalcommits.org/) format:

```
<type>[optional scope]: <description>
```

### Allowed Types:

- `feat`: A new feature
- `fix`: A bug fix
- `docs`: Documentation only changes
- `style`: Changes that do not affect the meaning of the code
- `refactor`: A code change that neither fixes a bug nor adds a feature
- `test`: Adding missing tests or correcting existing tests
- `chore`: Changes to build process or auxiliary tools
- `perf`: A code change that improves performance
- `ci`: Changes to CI configuration files and scripts
- `build`: Changes that affect the build system or external dependencies
- `revert`: Reverts a previous commit

### Examples:

✅ `feat: add cross-chain intent validation`
✅ `fix: resolve memory leak in prove method`
✅ `docs: update API documentation for routes`
✅ `refactor(inbox): simplify refund logic`
✅ `test: add edge case coverage for token transfers`
✅ `chore: update dependencies to latest versions`

### Important Notes:

- The description should be lowercase and not end with a period
- Use present tense ("add" not "added" or "adds")
- Scopes are optional but helpful for organizing changes
- If you need to bypass this check temporarily, add the label `skip-pr-title-check`

---

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test improvements
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] CI/Build changes

### CODE REVIEW

File: contracts/deposit/BaseDepositFactory.sol [hunk 1]
```diff
@@ -37,7 +37,6 @@ abstract contract BaseDepositFactory {
     error InvalidPortalAddress();
     error InvalidProverAddress();
     error InvalidDeadlineDuration();
-    error InvalidDestinationPortal();
```
Comment (fish-sammy) [inline contracts/deposit/BaseDepositFactory.sol:40]: Am I crazy or none of these errors are ever used not just this one?

**Review by fish-sammy — APPROVED** [review]
